elastic/kibana
# Remove unused code within the task manager polling section

This PR is a preparation to https://github.com/elastic/kibana/issues/151938 to avoid having a large PR to review all at once. Here, I'm removing support for on-demand polling (`pollRequests$`). This code path was used when we supported `runNow`, and is no longer used after transitioning to `runSoon`. There is a lot of underlying code related to this within the `task_poller.ts` file to support that capability, including a queue of events (combination of `pushOptionalIntoSet` and `pullFromSet`) before calling `concatMap` that is no longer needed.

## To verify

Use the following to log timestamps whenever task manager is polling for work
```
--- a/x-pack/plugins/task_manager/server/polling_lifecycle.ts
+++ b/x-pack/plugins/task_manager/server/polling_lifecycle.ts
@@ -243,6 +243,7 @@ export class TaskPollingLifecycle {
   }

   private pollForWork = async (): Promise<TimedFillPoolResult> => {
+    console.log('*** pollForWork', new Date().toISOString());
     return fillPool(
       // claim available tasks
       () => {
```

1. Ensure when Kibana is running, the pollForWork is logged on 3s intervals
2. Modify the code to slow down the updateByQuery to 5s, notice the next pollForWork isn't called until two update by queries have ran (main pool and reporting). Should be every 10s instead of 3s.

```
--- a/x-pack/plugins/task_manager/server/task_store.ts
+++ b/x-pack/plugins/task_manager/server/task_store.ts
@@ -444,6 +444,7 @@ export class TaskStore {
     // eslint-disable-next-line @typescript-eslint/naming-convention
     { max_docs: max_docs }: UpdateByQueryOpts = {}
   ): Promise<UpdateByQueryResult> {
+    await new Promise((resolve) => setTimeout(resolve, 5000));
     const { query } = ensureQueryOnlyReturnsTaskObjects(opts);
     try {
       const // eslint-disable-next-line @typescript-eslint/naming-convention
```

3. Slow down the `setTimeout` added in step two by setting it to `45000` and observe the pollForWork getting called on 30s interval with some timeout errors getting logged (same as main)
4. Undo the code changes from step two in favour of this one. Observe the pollForWork getting called on 3s intervals even though an error is thrown

```
--- a/x-pack/plugins/task_manager/server/task_store.ts
+++ b/x-pack/plugins/task_manager/server/task_store.ts
@@ -444,6 +444,7 @@ export class TaskStore {
     // eslint-disable-next-line @typescript-eslint/naming-convention
     { max_docs: max_docs }: UpdateByQueryOpts = {}
   ): Promise<UpdateByQueryResult> {
+    throw new Error('oh no!');
     const { query } = ensureQueryOnlyReturnsTaskObjects(opts);
     try {
       const // eslint-disable-next-line @typescript-eslint/naming-convention
```
